tests/sim/conversion_proxy.rs

```diff
@@ -188,12 +181,61 @@ fn test_transfer_with_wrong_currency() {
         ),
         deposit = transfer_amount
     );
-    assert_one_promise_error(result, "ERR_INVALID_ORACLE_RESPONSE");
+    assert_one_promise_error(
+        result,
+        "Only payments denominated in USD are implemented for now",
+    );
+}
+
+#[test]
+fn test_transfer_with_low_deposit() {
+    let (alice, bob, builder, proxy, _) = init();
+    let initial_alice_balance = alice.account().unwrap().amount;
+    let initial_bob_balance = bob.account().unwrap().amount;
+    let initial_contract_balance = proxy.account().unwrap().amount;
+    let transfer_amount = to_yocto("1000");
+    let payment_address = bob.account_id().try_into().unwrap();
+    let fee_address = builder.account_id().try_into().unwrap();
+
+    let result = call!(
+        alice,
+        proxy.transfer_with_reference(
+            PAYMENT_REF.into(),
+            payment_address,
+            U128::from(2000000),
+            USD.into(),
+            fee_address,
+            U128::from(0),
+            U64::from(0)
+        ),
+        deposit = transfer_amount
+    );
+    result.assert_success();
+    assert_eq!(result.logs().len(), 1, "Wrong number of logs");
+    assert!(result.logs()[0].contains("Deposit too small for payment"));
+
+    assert_eq!(
+        alice.account().unwrap().amount,
+        initial_alice_balance,
+        "Alice should not spend NEAR on a failed payment.",
+    );
+
+    // The contract's balance is slightly impacted by execution, hence we divide by 1 NEAR
```

Comment:
Is this comment correct? I don't see a divide by 1 NEAR in the logic below.

Reply:
Oh yes good point it does not apply anymore since the gas costs 0